projboard/tests/test_article.py

```diff
@@ -0,0 +1,148 @@
+import pytest
+from projboard.models.article import Article, Like, View, Subject, User
+
+
+TITLE = "Article Test"
+CONTENT = "I'm not gonna write all that!"
+
+
+class TestArticle:
+
+    @pytest.fixture
+    def sup(self):
+        return Subject(name='N')
+
+    @pytest.fixture
+    def sup1(self):
+        return Subject(name='R')
+
+    @pytest.fixture
+    def user(self):
+        return User(name="Anas", nickname="A99", email="Anas@gmail.com", password="Anas123")
+
+    @pytest.fixture
+    def art(self, user, sup):
+        return Article(user_id=user, title=TITLE, content=CONTENT, subject_id=sup)
+
+    # @pytest.mark.skip
+    def test_objectCreation(self, user, art):
+        assert art.title == TITLE
+        assert art.content == CONTENT
+        # testing __str__ function
+        assert art.__str__() == art.title
+
+    # @pytest.mark.skip
+    def test_PartFunctions(self, art):
+        # testing @property functions => PartOf..()
+        # whent title & content length is less than reality
+        assert art.PartOfTitle(5) == art.title[:5] + '...'
+        assert art.PartOfContent(5) == art.content[:5] + '...'
+        # whent title & content length is greater than reality
+        assert art.PartOfTitle(100) == art.title[:100]
+        assert art.PartOfContent(100) == art.content[:100]
+
+    # @pytest.mark.skip
+    @pytest.mark.django_db
+    def test_objectSavedInDb(self, user, art, sup):
+        sup.save()
+        user.save()
+        art.save()
+        assert art in Article.objects.all()
+
+    # @pytest.mark.skip
+    @pytest.mark.django_db
+    # using Edite function
+    def test_EditeObject(self, user, art, sup, sup1):
+        sup.save()
```

Comment:
This should move into fixures
You can return an array of subjects see previous comments

Reply:
done